alonzo/impl/src/Cardano/Ledger/Alonzo/TxInfo.hs

```diff
@@ -257,8 +259,11 @@ getWitVKeyHash = P.PubKeyHash . fromShort . (\(UnsafeHash x) -> x) . (\(KeyHash
 transDataPair :: (DataHash c, Data era) -> (P.DatumHash, P.Datum)
 transDataPair (x, y) = (transDataHash' x, P.Datum (getPlutusData y))
 
-transCostModel :: CostModel -> P.CostModelParameters
-transCostModel (CostModel mp) = Map.foldlWithKey' (\ans bytes n -> Map.insert (show bytes) n ans) Map.empty mp
+transCostModel :: CostModel -> P.CostModel
+transCostModel (CostModel mp) =
+  case P.validateAndCreateCostModel mp of
+    Nothing -> P.defaultCostModel -- TODO validation should be before this
```

Comment:
This is definitely not what we want in the long term. We need a plan for where and how to handle the cost model validation on the ledger side, which is probably our next priority.